bundles/automation/org.eclipse.smarthome.automation.module.media/src/main/java/org/eclipse/smarthome/automation/module/media/internal/MediaActionTypeProvider.java

```diff
@@ -0,0 +1,89 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.automation.Visibility;
+import org.eclipse.smarthome.automation.module.media.handler.PlayActionHandler;
+import org.eclipse.smarthome.automation.type.ActionType;
+import org.eclipse.smarthome.automation.type.Input;
+import org.eclipse.smarthome.automation.type.ModuleType;
+import org.eclipse.smarthome.automation.type.ModuleTypeProvider;
+import org.eclipse.smarthome.automation.type.Output;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter.Type;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameterBuilder;
+import org.eclipse.smarthome.config.core.ParameterOption;
+import org.eclipse.smarthome.core.audio.AudioManager;
+
+/**
+ * This class dynamically provides the Play action type.
+ * This is necessary since there is no other way to provide dynamic config param options for module types.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class MediaActionTypeProvider implements ModuleTypeProvider {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public ModuleType getModuleType(String UID, Locale locale) {
+        if ("PlayAction".equals(UID)) {
+            return getPlayActionType(locale);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Collection<ModuleType> getModuleTypes(Locale locale) {
+        return Collections.singleton(getPlayActionType(locale));
+    }
+
+    private ModuleType getPlayActionType(Locale locale) {
+        return new ActionType("PlayAction", getConfigDesc(), "Play Sound", "Plays a sound file", null,
+                Visibility.VISIBLE, new ArrayList<Input>(), new ArrayList<Output>());
+    }
+
+    private List<ConfigDescriptionParameter> getConfigDesc() {
+        ConfigDescriptionParameter param = ConfigDescriptionParameterBuilder
+                .create(PlayActionHandler.PARAM_SOUND, Type.TEXT).withRequired(true).withLabel("Sound")
+                .withDescription("The sound to play").withOptions(getOptions()).build();
+        return Collections.singletonList(param);
+    }
+
+    /**
+     * This method creates one option for every file that is found in the sounds directory.
+     * as a label, the file extension is removed and the string is capitalized.
+     *
+     * @return a list of parameter options representing the sound files
+     */
+    private List<ParameterOption> getOptions() {
+        List<ParameterOption> options = new ArrayList<>();
+        File soundsDir = Paths.get(ConfigConstants.getConfigFolder(), AudioManager.SOUND_DIR).toFile();
+        if (soundsDir.isDirectory()) {
+            for (String fileName : soundsDir.list()) {
+                if (fileName.contains(".")) {
+                    String soundName = StringUtils.capitalize(fileName.substring(0, fileName.indexOf(".")));
```

Comment:
I think most solutions will decide which formats they (and their sinks) support, so from a (dumb) user perspective I think the format should not be displayed in this list.
What I would like to add in future is another parameter for the sink to use (if not set, the default is used). This could then be taken into account when creating the option list, e.g. filtering out mp3 if the sink does not support it.